### PR TITLE
add cls to e2e

### PR DESCRIPTION
**Pull request by zhangjunlongtech**
Thank you for your contribution to the MindOCR repo.
Before submitting this PR, please make sure:

- [ ] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [ ] Your code builds clean without any errors or warnings
- [ ] You are using approved terminology
- [ ] You have added unit tests

## Motivation

add cls to e2e

## Test Plan
测试包含cls的predict_system e2e
(CAN) [root@57d7ec97e0a0 mindocr_dot]# python /home/nginx/work/zhangjunlong/mindocr_dot/tools/infer/text/predict_system.py --image_dir /home/nginx/work/zhangjunlong/mindocr_cls/tests/st/test_cls/img_for_e2e/example_img04_for_e2e.png --det_algorithm DB++  --rec_algorithm CRNN --cls_algorithm M3

[2024-12-20 17:32:48] mindocr.models.backbones.mindcv_models.utils INFO - Finish loading model checkpoint from: /root/.mindspore/models/dbnetpp_resnet50-068166c2.ckpt
[2024-12-20 17:32:48] mindocr INFO - Init detection model: DB++ --> dbnetpp_resnet50. Model weights loaded from pretrained url
[2024-12-20 17:32:48] mindocr INFO - Pick optimal preprocess hyper-params for det algo DB++:
 {'DetResize': {'target_size': None, 'keep_ratio': True, 'limit_side_len': 960, 'limit_type': 'max', 'padding': False, 'force_divisable': True}}
[2024-12-20 17:32:48] mindocr.data.transforms.det_transforms INFO - `limit_type` is max. Image will be resized by limiting the max side length to 960.
[2024-12-20 17:32:48] mindocr INFO - recognize in batch mode batch_size: 8
[2024-12-20 17:32:49] mindocr.models.backbones.mindcv_models.utils INFO - Finish loading model checkpoint from: /root/.mindspore/models/crnn_resnet34-83f37f07.ckpt
[2024-12-20 17:32:49] mindocr INFO - Init recognition model: CRNN --> crnn_resnet34. Model weights loaded from pretrained url
[2024-12-20 17:32:49] mindocr WARNING - `rec_image_shape` [' 32', ' 320'] dose not meet the network input requirement or is not optimal, which should be [32, 100] under batch mode = True
[2024-12-20 17:32:49] mindocr INFO - Pick optimal preprocess hyper-params for rec algo CRNN:
target_height:  32
target_width:   100
padding:        False
keep_ratio:     False
norm_before_pad:        False
[2024-12-20 17:32:49] mindocr.postprocess.rec_postprocess INFO - `character_dict_path` for RecCTCLabelDecode is not given. Default dict "0123456789abcdefghijklmnopqrstuvwxyz" is applied. Only number and English letters (regardless of lower/upper case) will be recognized and evaluated.
[2024-12-20 17:32:49] mindocr INFO - classify in batch mode batch_size: 8
[2024-12-20 17:32:50] mindocr.models.backbones.mindcv_models.utils INFO - Finish loading model checkpoint from: /root/.mindspore/models/mobilenet_v3_small_100-c884b105.ckpt
[2024-12-20 17:32:50] mindocr.models.backbones.mindcv_models.utils INFO - Finish loading model checkpoint from: /root/.mindspore/models/cls_mobilenetv3-92db9c58.ckpt
[2024-12-20 17:32:50] mindocr INFO - Init classification model: M3 --> cls_mobilenet_v3_small_100_model. Model weights loaded from pretrained url
[2024-12-20 17:32:50] mindocr INFO - 
INFO: Infering [1/1]: /home/nginx/work/zhangjunlong/mindocr_cls/tests/st/test_cls/img_for_e2e/example_img04_for_e2e.png
[2024-12-20 17:32:50] mindocr INFO - Original image shape: (350, 350, 3)
[2024-12-20 17:32:50] mindocr INFO - After det preprocess: (3, 352, 352)
[2024-12-20 17:32:59] mindocr INFO - Num detected text boxes: 7
Det time: 8.891462087631226
[2024-12-20 17:32:59] mindocr INFO - num images for cls: 7
[2024-12-20 17:32:59] mindocr INFO - CLS img idx range: [0, 7)
[2024-12-20 17:33:05] mindocr INFO - The number of images corrected by rotation is 6/7
CLS time: 5.622709274291992
[2024-12-20 17:33:05] mindocr INFO - num images for rec: 7
[2024-12-20 17:33:05] mindocr INFO - Rec img idx range: [0, 7)
[2024-12-20 17:33:12] mindocr INFO - Recognized texts: 
warning 1.0
prevent 0.9961774945259094
snoles  0.9545091986656189
injury  0.9999916553497314
read    1.0
the     1.0
manual  0.9999873042106628
Rec time: 7.001410484313965
[2024-12-20 17:33:12] mindocr INFO - Total time:21.51879096031189
[2024-12-20 17:33:12] mindocr INFO - Average FPS: 0.04647101232798565
[2024-12-20 17:33:12] mindocr INFO - Averge time cost: {'det': 8.891462087631226, 'cls': 5.622709274291992, 'rec': 7.001410484313965, 'all': 21.51879096031189}
[2024-12-20 17:33:12] mindocr INFO - Done! Results saved in ./inference_results/system_results.txt

测试不包含cls的predict_system e2e
(CAN) [root@57d7ec97e0a0 mindocr_dot]# python /home/nginx/work/zhangjunlong/mindocr_dot/tools/infer/text/predict_system.py --image_dir /home/nginx/work/zhangjunlong/mindocr_cls/tests/st/test_cls/img_for_e2e/example_img03_for_e2e.png --det_algorithm DB++  --rec_algorithm CRNN

[2024-12-20 17:33:43] mindocr.models.backbones.mindcv_models.utils INFO - Finish loading model checkpoint from: /root/.mindspore/models/dbnetpp_resnet50-068166c2.ckpt
[2024-12-20 17:33:43] mindocr INFO - Init detection model: DB++ --> dbnetpp_resnet50. Model weights loaded from pretrained url
[2024-12-20 17:33:43] mindocr INFO - Pick optimal preprocess hyper-params for det algo DB++:
 {'DetResize': {'target_size': None, 'keep_ratio': True, 'limit_side_len': 960, 'limit_type': 'max', 'padding': False, 'force_divisable': True}}
[2024-12-20 17:33:43] mindocr.data.transforms.det_transforms INFO - `limit_type` is max. Image will be resized by limiting the max side length to 960.
[2024-12-20 17:33:43] mindocr INFO - recognize in batch mode batch_size: 8
[2024-12-20 17:33:44] mindocr.models.backbones.mindcv_models.utils INFO - Finish loading model checkpoint from: /root/.mindspore/models/crnn_resnet34-83f37f07.ckpt
[2024-12-20 17:33:44] mindocr INFO - Init recognition model: CRNN --> crnn_resnet34. Model weights loaded from pretrained url
[2024-12-20 17:33:44] mindocr WARNING - `rec_image_shape` [' 32', ' 320'] dose not meet the network input requirement or is not optimal, which should be [32, 100] under batch mode = True
[2024-12-20 17:33:44] mindocr INFO - Pick optimal preprocess hyper-params for rec algo CRNN:
target_height:  32
target_width:   100
padding:        False
keep_ratio:     False
norm_before_pad:        False
[2024-12-20 17:33:44] mindocr.postprocess.rec_postprocess INFO - `character_dict_path` for RecCTCLabelDecode is not given. Default dict "0123456789abcdefghijklmnopqrstuvwxyz" is applied. Only number and English letters (regardless of lower/upper case) will be recognized and evaluated.
[2024-12-20 17:33:44] mindocr INFO - 
INFO: Infering [1/1]: /home/nginx/work/zhangjunlong/mindocr_cls/tests/st/test_cls/img_for_e2e/example_img03_for_e2e.png
[2024-12-20 17:33:44] mindocr INFO - Original image shape: (350, 350, 3)
[2024-12-20 17:33:44] mindocr INFO - After det preprocess: (3, 352, 352)
[2024-12-20 17:33:52] mindocr INFO - Num detected text boxes: 7
Det time: 8.783303022384644
[2024-12-20 17:33:52] mindocr INFO - num images for rec: 7
[2024-12-20 17:33:52] mindocr INFO - Rec img idx range: [0, 7)
[2024-12-20 17:33:59] mindocr INFO - Recognized texts: 
manual  0.9541060328483582
read    1.0
the     0.8334037661552429
injury  1.0
serious 1.0
prevent 1.0
warning 1.0
Rec time: 6.477664470672607
[2024-12-20 17:33:59] mindocr INFO - Total time:15.263165473937988
[2024-12-20 17:33:59] mindocr INFO - Average FPS: 0.065517208845538
[2024-12-20 17:33:59] mindocr INFO - Averge time cost: {'det': 8.783303022384644, 'rec': 6.477664470672607, 'all': 15.263165473937988}
[2024-12-20 17:33:59] mindocr INFO - Done! Results saved in ./inference_results/system_results.txt

### CODE REVIEW

**Review by can-gaa-hou — APPROVED** [review]
lgtm
